app/models/visit.rb

```diff
@@ -12,7 +12,10 @@ class Visit
   attribute :vo_number, String
 
   validates_presence_of :visit_id
-  validates_size_of :slots, within: 1..MAX_SLOTS, on: :date_and_time, message: 'must be at least one and at most three'
+  validates_size_of :slots,
+    within: 1..MAX_SLOTS,
+    on: :date_and_time,
+    message: 'must be at least one and at most three'
```

Comment:
@threedaymonk :+1: The `on: :date_and_time` I believe it is necessary, because without it we are having some failing test:

```ruby
rspec './spec/features/visitor_information_page_spec.rb[1:2:5:1]' # visitor enters visitor information and they fill out all fields for 4 additional visitors displays the calendar
rspec './spec/features/visitor_information_page_spec.rb[1:2:4:1]' # visitor enters visitor information and they fill out all fields for 3 additional visitors displays the calendar
rspec './spec/features/visitor_information_page_spec.rb[1:2:2:1]' # visitor enters visitor information and they fill out all fields for 1 additional visitors displays the calendar
rspec ./spec/features/visitor_information_page_spec.rb:25 # visitor enters visitor information and they fill out all fields for one visitor displays the calendar
rspec './spec/features/visitor_information_page_spec.rb[1:2:3:1]' # visitor enters visitor information and they fill out all fields for 2 additional visitors displays the calendar
rspec './spec/features/visitor_information_page_spec.rb[1:2:6:1]' # visitor enters visitor information and they fill out all fields for 5 additional visitors displays the calendar
rspec ./spec/features/unexpected_journey_spec.rb:72 # unexpected journeys through the application session expires on the confirmation page
rspec ./spec/features/unexpected_journey_spec.rb:56 # unexpected journeys through the application session expires while entering visit details
rspec ./spec/controllers/visitors_details_controller_spec.rb:52 # VisitorsDetailsController given valid visitor information updates visitor information
rspec './spec/controllers/visitors_details_controller_spec.rb[1:4:4:1:1]' # VisitorsDetailsController behaves like a visitor data manipulator with invalid data given only a single child visitor when the prison assumes each adult to be 18 or older allows the visitor to proceed
rspec './spec/controllers/visitors_details_controller_spec.rb[1:4:4:2:1]' # VisitorsDetailsController behaves like a visitor data manipulator with invalid data given only a single child visitor when the prison assumes each adult to be some other age allows the visitor to proceed
rspec ./spec/models/visit_spec.rb:41 # Visit is valid with three or fewer adult visitors
rspec ./spec/models/visit_spec.rb:16 # Visit restricts the number of visitors
rspec ./spec/models/visit_spec.rb:56 # Visit given a prison which treats a child as an adult for seating purposes requires at least one real adult
rspec ./spec/features/datepicker_spec.rb:36 # visitor selects a date Choosing a date more than 14 days from now
rspec ./spec/features/datepicker_spec.rb:41 # visitor selects a date Booking a valid slot
rspec ./spec/features/datepicker_spec.rb:30 # visitor selects a date Choosing a date more than 3 working days from now
rspec ./spec/features/datepicker_spec.rb:24 # visitor selects a date Choosing a date in the past
```

Also I found that is using it [slots_manipulator_line 18](https://github.com/ministryofjustice/prison-visits/blob/master/lib/slots_manipulator.rb#L18), [visit_spec_line 30](https://github.com/ministryofjustice/prison-visits/blob/master/spec/models/visit_spec.rb#L30),  [visit_spec_line 34](https://github.com/ministryofjustice/prison-visits/blob/master/spec/models/visit_spec.rb#L34) and  [visit_spec_line 38](https://github.com/ministryofjustice/prison-visits/blob/master/spec/models/visit_spec.rb#L38) if I understand correct.

By refactoring it to 
```ruby
validates :slots, length: {
    in: 1..MAX_SLOTS,
    on: :date_and_time
  }
```
all the test are passing.